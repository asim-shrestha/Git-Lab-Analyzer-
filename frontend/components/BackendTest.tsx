import React, {useState, useEffect} from "react";
import axios, {AxiosError, AxiosResponse} from "axios";
import { useSnackbar } from 'notistack';

const BackendTest = () => {
    const { enqueueSnackbar } = useSnackbar();
    const [repos, setRepos] = useState([]);

    // Load repo data
    useEffect(() => {
        axios
            .get(`${process.env.NEXT_PUBLIC_API_URL}/projects`)
            .then((resp: AxiosResponse) => {
                setRepos(resp.data);
                enqueueSnackbar(`Successfully connected`, {variant: 'success',});
            })
            .catch((err: AxiosError) => {
                enqueueSnackbar(`Failed to connect: ${err.message}`, {variant: 'error',});
            });
    }, []);

    return (
        <div>
            <h4>API URL: {process.env.NEXT_PUBLIC_API_URL}</h4>
            <p>API data: {JSON.stringify(repos)}</p>
        </div>
    );
};

export default BackendTest;
