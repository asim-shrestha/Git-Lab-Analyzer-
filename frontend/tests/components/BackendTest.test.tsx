import React from 'react';
import BackendTest from '../../components/BackendTest';
import {render} from "@testing-library/react";

describe("BackendTest", () =>{
    const mockAxios = jest.spyOn(require('axios'), 'get');
    const mockUseEffect = jest.spyOn(React, 'useEffect');
    const mockEnqueue = jest.spyOn(require('notistack'), "useSnackbar");
    let enqueueSnackbar = jest.fn();

    beforeAll(() =>{
        mockEnqueue.mockImplementation(() => {return {enqueueSnackbar}});

    })

    it("Snapshot BackendTest", () => {
        const { container } = render(
            <BackendTest />
        )
        expect(container).toMatchSnapshot();

    })
    it('Test useEffect', ()=> {
        expect(mockUseEffect).toBeCalled();
    })

    it('Test axios',()=> {
        expect(mockAxios).toHaveBeenCalledTimes(1);
    })
})