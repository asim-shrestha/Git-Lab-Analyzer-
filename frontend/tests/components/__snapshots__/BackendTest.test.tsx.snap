// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackendTest Snapshot BackendTest 1`] = `
<div>
  <div>
    <h4>
      API URL: 
    </h4>
    <p>
      API data: 
      []
    </p>
  </div>
</div>
`;
